web3swift/src/Client/EthereumWebSocketClient.swift

```diff
@@ -0,0 +1,1198 @@
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Int, Error>) in
+            eth_getTransactionCount(address: address, block: block, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_getTransaction(byHash txHash: String) async throws -> EthereumTransaction {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<EthereumTransaction, Error>) in
+            eth_getTransaction(byHash: txHash, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_getTransactionReceipt(txHash: String) async throws -> EthereumTransactionReceipt {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<EthereumTransactionReceipt, Error>) in
+            eth_getTransactionReceipt(txHash: txHash, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_getLogs(addresses: [EthereumAddress]?, topics: [String?]?, fromBlock from: EthereumBlock = .Earliest, toBlock to: EthereumBlock = .Latest) async throws -> [EthereumLog] {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<[EthereumLog], Error>) in
+            eth_getLogs(addresses: addresses, topics: topics, fromBlock: from, toBlock: to, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_getLogs(addresses: [EthereumAddress]?, orTopics topics: [[String]?]?, fromBlock from: EthereumBlock = .Earliest, toBlock to: EthereumBlock = .Latest) async throws ->  [EthereumLog] {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<[EthereumLog], Error>) in
+            eth_getLogs(addresses: addresses, orTopics: topics, fromBlock: from, toBlock: to, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_getBlockByNumber(_ block: EthereumBlock) async throws -> EthereumBlockInfo {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<EthereumBlockInfo, Error>) in
+            eth_getBlockByNumber(block, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_call(_ transaction: EthereumTransaction, block: EthereumBlock = .Latest) async throws -> String {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<String, Error>) in
+            eth_call(transaction, block: block, completionHandler: continuation.resume)
+        }
+    }
+
+    public func eth_call(_ transaction: EthereumTransaction, resolution: CallResolution = .noOffchain(failOnExecutionError: true), block: EthereumBlock = .Latest) async throws -> String {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<String, Error>) in
+            eth_call(transaction, resolution: resolution, block: block, completionHandler: continuation.resume)
+        }
+    }
```

Comment:
Lots of duplicated code given we reimplement with only the request transport side being different. Maybe reworking the WSS to be the transport layer instead of a full client implementation could work better? It would require lot more changes though